src/sparse_approximations.jl

```diff
@@ -281,7 +281,12 @@ function _compute_intermediates(fx::FiniteGP, y::AbstractVector{<:Real}, fz::Fin
 end
 
 function consistency_check(fx, y)
-    @assert length(fx) == length(y)
+    length(fx) == length(y) || throw(
+        DimensionMismatch(
+            "length(fx) = $(length(fx)) and length(y) = $(length(y)) must match"
```

Comment:
Could we use a clearer error message? Probably users won't know what `fx` and `y` are.

Reply:
it's the same variable names as the arguments in the docstrings of `elbo` and `dtc`, but how about
```suggestion
            "The FiniteGP's length(fx) = $(length(fx)) and the targets's length(y) = $(length(y)) must match"
```
?

Reply:
That's not guaranteed though (we could change them, use the check in other fumctions etc.), and even if the arguments the same names users might not be aware of these names. Therefore I don't think the variable names are helpful at all and suggest removing them completely. Maybe just use something like
```julia
"the dimension of the projected GP ($(length(fx))) and the number of targets ($(length(y))) must be equal"
```

Reply:
If you've got such concrete ideas/thoughts on it, why don't you just go and make it as a code suggestion?

Reply:
Actually I wanted to make one but I didn't have time to deal with the inconveniences of the Github app on my phone. It doesn't allow me to add suggestions in follow-up comments via the interface but requires manually copying and modifying the code. I can add a suggestion this evening/night when I have access to my computer.

Reply:
fair enough. how about:
```suggestion
            "the dimension of the projected GP (here: $(length(fx))) must equal the number of targets (here: $(length(y)))"
```